tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1120 @@
+        4.  Create second matcher for some attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        5.  Go to YAML output of associated host.
+
+        @assert: The YAML output has the value only for fqdn matcher.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_attribute_priority(self):
+        """Matcher Value set on Attribute Priority for Host - alternate priority.
+
+        @id: 7e52b054-4fcb-4c58-ae49-0d3348d14570
+
+        @steps:
+
+        1.  Create variable with some default value.
+        2.  Set some attribute(other than fqdn) as top priority attribute.
+        Note - The fqdn/host should have this attribute.
+        3.  Create first matcher for fqdn with valid details.
+        4.  Create second matcher for attribute of step 3 with valid details.
+        5.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the value only for step 5 matcher.
+        2.  The YAML output doesn't have value for fqdn/host matcher.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_merge_override(self):
+        """Merge the values of all the associated matchers.
+
+        @id: b9c9b1c7-ff9a-4080-aeee-3b61b5414332
+
+        @steps:
+
+        1.  Create variable with some default value.
+        2.  Create first matcher for attribute fqdn with valid details.
+        3.  Create second matcher for other attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        4.  Create more matchers for some more attributes if any.
+        Note - The fqdn/host should have this attributes.
+        5.  Set --merge-overrides to true.
+        6.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all the associated
+        matchers.
+        2.  The YAML output doesn't have the default value of variable.
+        3.  Duplicate values in YAML output if any are displayed.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_merge_override(self):
+        """Test merge the values from non associated matchers.
+
+        @id: 3cc2a7b3-7b46-4c8c-b719-79c004ae04c6
+
+        @steps:
+
+        1.  Create variable with some default value.
+        3.  Create first matcher for attribute fqdn with valid details.
+        4.  Create second matcher for other attribute with valid details.
+        Note - The fqdn/host should not have this attribute.
+        5.  Create more matchers for some more attributes if any.
+        Note - The fqdn/host should not have this attributes.
+        6.  Set --merge-overrides to true.
+        7.  Go to YAML output of associated host.
```

Comment:
We don't have yaml output in CLI, even if you specify --output-format=yaml, it will not be the same as we have for UI